### ISSUE TITLE
🐛 [Bug] truncate_long_and_double not used

### ISSUE THREAD

**Issue by GiteZz** [NVIDIA/Torch-TensorRT#841]
##  Bug Description

I'm trying to use the library to convert the Detectron2 Mask-RCNN model to TensorRT. For that I first trace the model with Jit and then use the torch_tensorrt.compile function. During the compilation I get this error: 

`RuntimeError: [Error thrown at core/partitioning/shape_analysis.cpp:116] Unable to process subgraph input type of at::kLong/at::kDouble, try to compile model with truncate_long_and_double enabled`

However in my compile command I set `truncate_long_and_double` to `True`. I was hoping that it might fall back to torch when encountering such an error, but that is not the case.

## To Reproduce

Below is the full code that I use for compiling the model. 
```
import torch
import torch_tensorrt

from model import get_model
from pathlib import Path
from detectron2.export.flatten import TracingAdapter

trt_input = torch.randn((3, 1330, 1330))

def inference_func(model, image):
    inputs = [{"image": image}]
    return model.inference(inputs, do_postprocess=False)[0]

compile_settings = {
    "inputs": [torch_tensorrt.Input([1, 3, 1330, 1330], dtype=torch_tensorrt.dtype.float32)],
    "require_full_compilation": False,
    # "enabled_precisions": {torch.half},  # Run with FP16
    "truncate_long_and_double": True,
}

model = get_model(Path.cwd() / "model_files")


wrapper = TracingAdapter(model.model, trt_input, inference_func)
wrapper.eval()

traced_model = torch.jit.trace(wrapper, trt_input)
traced_model.save("traced_model.pt")

loaded = torch.jit.load('traced_model.pt')

trt_model = torch_tensorrt.compile(loaded, **compile_settings)
```

The model in question is the `COCO-InstanceSegmentation/mask_rcnn_R_50_FPN_3x.yaml` model from the Detectron2 library. This is the full traceback:

```
Serialized   File "code/__torch__/detectron2/modeling/poolers.py", line 26
    sizes = torch.to(ret, dtype=4, layout=0, device=torch.device("cuda:0"))
    _6 = torch.arange(1, dtype=6, layout=0, device=torch.device("cuda:0"), pin_memory=False)
    indices = torch.repeat_interleave(_6, sizes)
              ~~~~~~~~~~~~~~~~~~~~~~~ <--- HERE
    _7 = torch.slice(indices, 0, 0, 9223372036854775807)
    pooler_fmt_boxes = torch.cat([torch.unsqueeze(_7, 1), _4], 1)

WARNING: [Torch-TensorRT] - Input type for doing shape analysis could not be determined, defaulting to F32
Traceback (most recent call last):
  File "/workspaces/ROS_Instance_Segmentation/trt.py", line 40, in <module>
    trt_model = torch_tensorrt.compile(loaded, **compile_settings)
  File "/usr/local/lib/python3.8/dist-packages/torch_tensorrt/_compile.py", line 97, in compile
    return torch_tensorrt.ts.compile(ts_mod, inputs=inputs, enabled_precisions=enabled_precisions, **kwargs)
  File "/usr/local/lib/python3.8/dist-packages/torch_tensorrt/ts/_compiler.py", line 119, in compile
    compiled_cpp_mod = _C.compile_graph(module._c, _parse_compile_spec(spec))
RuntimeError: [Error thrown at core/partitioning/shape_analysis.cpp:116] Unable to process subgraph input type of at::kLong/at::kDouble, try to compile model with truncate_long_and_double enabled
```

## Expected behavior

Fall back to pytorch or casting to correct type.

## Environment

> Build information about Torch-TensorRT can be found by turning on debug messages

 - Torch-TensorRT Version: 1.0.0
 - PyTorch Version: 1.10.2+cu113
 - CPU Architecture: x86-64
 - OS (e.g., Linux): nvidia/cuda:11.3.1-cudnn8-devel-ubuntu20.04 docker on Ubuntu 20.04 base OS
 - How you installed PyTorch (`conda`, `pip`, `libtorch`, source): `pip3 install torch==1.10.2+cu113 -f https://download.pytorch.org/whl/cu113/torch_stable.html`
 - Build command you used (if compiling from source): /
 - Are you using local sources or building from archives: /
 - Python version: 3.8
 - CUDA version: 11.3.1
 - GPU models and configuration: RTX3000
 - Any other relevant information: /

**Comment by narendasan** [comment]
Able to replicate this. Looking into root cause

**Comment by yuezhuang1387** [comment]
> Able to replicate this. Looking into root cause

I have encountered the same problem

**Comment by narendasan** [comment]
This was solved with #868
